docs/developers/sc-calls-format.md

```diff
@@ -0,0 +1,344 @@
+if err != nil {
+	return err
+}
+addressObj := data.NewAddressFromBytes(addressBytes)
+
+fmt.Println(addressObj.AddressAsBech32String())
+```
+
+will output `erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th`.
+
+### Converting addresses using erdjava
+
+Find more about `erdjava` [here](/sdk-and-tools/erdjava/).
+
+```java
+System.out.println(Address.fromBech32("erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th").hex());
+```
+
+will output `0139472eff6886771a982f3083da5d421f24c29181e63888228dc81ca60d69e1`.
+
+Alternatively, hex addresses can be converted to bech32 as follows:
+
+```java
+System.out.println(Address.fromHex("0139472eff6886771a982f3083da5d421f24c29181e63888228dc81ca60d69e1").bech32());
+```
+
+will output `erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th`.
+
+## Converting string values
+
+For situations when a string argument is desired for a smart contract call, it can be simply obtained by using 
+built-in libraries to convert them into hexadecimal format.
+
+:::important
+Make sure that the result has an even number of characters.
+:::
+
+Below you can find some examples:
+
+:::note
+By no means, these code snippets don't provide a coding guideline, but rather they are simple examples on how to perform the necessary actions.
+:::
+
+### Examples
+
+string --> hex
+```
+ok          --> 6f6b
+MEX-455c57  --> 4d45582d343535633537
+```
+
+### Converting string values in javascript
+
+```
+console.log(Buffer.from("ok").toString("hex"));  // 6f6b
+```
+
+for converting hex-encoded string to regular string:
+
+```
+console.log(Buffer.from("6f6b", "hex").toString()); // ok
+```
+
+### Converting string values in java
+```
+String inputHex = Hex.encodeHexString("ok".getBytes(StandardCharsets.UTF_8));
+if (inputHex.length() % 2 != 0) {
+   inputHex = "0" + inputex;
+}
+
+System.out.println(inputHex);  // 6f6b
+```
+
+for converting hex-encoded string to regular string:
+
+```
+byte[] bytes = Hex.decodeHex("6f6b".toCharArray());
+
+String result = new String(bytes, StandardCharsets.UTF_8); // ok
+```
+
+### Converting string values in go
+
+```
+fmt.Println(hex.EncodeToString([]byte("ok"))) // 6f6b
+```
+
+for converting hex-encoded string to regular string:
+
+```
+decodedInput, err := hex.DecodeString("6f6b")
+if err != nil {
+    return err
+}
+
+fmt.Println(string(decodedInput)) // ok
+```
+
+## Converting numeric values
+
+For situations when a numeric argument is desired for a smart contract call, it can be simply obtained by using
+built-in libraries to convert them into hexadecimal format.
+
+:::important
+Make sure that the result has an even number of characters.
+:::
+
+Below you can find some examples. They use big integer / number libraries to ensure the code works for large values as well:
+
+:::note
+By no means, these code snippets don't provide a coding guideline, but rather they are simple examples on how to perform the necessary actions.
```

Comment:
I'm not sure whether the negation "by no means" requires a negation or a non-negation afterwards (don't provide vs. do provide). Perhaps rephrase a bit?

Reply:
rephrased